elixir/apps/domain/lib/domain/auth.ex

```diff
@@ -164,29 +174,24 @@ defmodule Domain.Auth do
   The token is encoded as a tuple of {account_id, token_id, secret_fragment}
   which is then signed using Plug.Crypto to ensure it can't be tampered with.
   """
-  def encode_fragment!(%Token{secret_fragment: fragment, type: type} = token)
-      when not is_nil(fragment) do
-    body = {token.account_id, token.id, fragment}
-    config = fetch_config!()
-    key_base = Keyword.fetch!(config, :key_base)
-    salt = Keyword.fetch!(config, :salt)
-    "." <> Plug.Crypto.sign(key_base, salt <> to_string(type), body)
-  end
+  def encode_fragment!(%Token{type: :client, secret_nonce: nonce} = token),
+    do: encode_token(token.account_id, token.id, token.secret_fragment, nonce, "client")
 
-  def encode_fragment!(%Domain.RelayToken{} = token) do
-    encode_infrastructure_token(token, "relay", nil)
-  end
+  def encode_fragment!(%Domain.RelayToken{} = token),
+    do: encode_token(nil, token.id, token.secret_fragment, "", "relay")
```

Comment:
Somewhat in the same vein as the comment above about the default value, what would you think about doing something like this?:


```suggestion
    do
      nonce = ""
      encode_token(nil, token.id, token.secret_fragment, nonce, "relay")
    end
```

Reply:
I was just thinking

```
def encode_token(account_id, token.id, token.secret_fragment, type, nonce \\ "")
```

Reply:
That works too

Reply:
fixed